src/compiler/compile/render_dom/wrappers/Window.ts

```diff
@@ -127,7 +127,11 @@ export default class WindowWrapper extends Wrapper {
 
 			component.partly_hoisted.push(b`
 				function ${id}() {
-					${props.map(prop => b`$$invalidate('${prop.name}', ${prop.name} = @_window.${prop.value});`)}
+					${props.map(prop =>
```

Comment:
If possible, it would be good to use the `Component#invalidate()` helper to generate this. It has logic built into it to `.set()` on stores and `$$invalidate()` for non-stores. Same below with the `online` binding.